### ISSUE TITLE
Stats within WP dashboard

### ISSUE THREAD

**Issue by metmarkosaric** [plausible/wordpress#17]
We now have the stats API ready so wondering if that makes it possible for us to include some of the stats within the WP dashboard? Or does it make the most sense to wait for [the embedded mode](https://github.com/plausible/analytics/discussions/213) and then simply embed the complete dashboard? What is the best solution for WordPress?

Stats API docs: https://plausible.io/docs/stats-api
[Example](https://twitter.com/jackabox/status/1363973796097622020) of a similar thing done with our API for Statamic:

![Eu3O6yzXAAYqJNX](https://user-images.githubusercontent.com/34340819/108814672-81127b80-75b3-11eb-9515-3dfc77b6320f.jpg)

**Comment by ukutaht** [comment]
This will be announced tomorrow but I deployed support for iFrame embed to prod today. Here are the instructions: https://plausible.io/docs/embed-dashboard

Embedding is probably the easiest way to show the whole dashboard along with filters and other interactive behaviour that we have built in the plausible web app.

To make the WP experience even better it would be great to be able to generate a shared link through the API. Then the plugin could provision the shared link and embed it without manual steps by the user. Currently, the user will need to create a shared link on our end and enter it in their WP dashboard.

**Comment by LittleBigThing** [comment]
Hi,

Just for some more background information around implementing this for the WordPress plugin.

As a (freelance sustainable privacy-conscious :-)) web designer/developer, I have a Plausible subscription that I use for all my clients who want to track their site stats. This is how it evolved for me to provide them the data:

1. I subscribed them to the monthly (or weekly) email report
2. I provided them a shared link, sometimes with password protection, when that became available
3. I wrote a small plugin that allows (me) to paste the shared link in WP, save it there and ‘magically’ enable them viewing the stats in their WP dashboard via an iframe embed.

![plausible dashboard without link](https://user-images.githubusercontent.com/5901923/111861891-f292df00-8951-11eb-987b-452d8569df34.png)

![plausible dashboard with link](https://user-images.githubusercontent.com/5901923/111862369-418e4380-8955-11eb-8390-9ff423f000a6.png)

I like this latter approach: the stats are always there in their dashboard, no more lost shared links. :-)

This is a kind of controlled way that works fine in my situation. Since clients don’t have access to my Plausible account, they cannot generate their own shared link. In my case, they don't have complete admin access to their WP dashboard either, but that is often the case for others‘ setups as far as I hear (I don't do it because it is not sustainable in my opinion and in my setup).

How would this work if the API was used to generate a shared link via the plugin? There is currently no authentication needed for the Plausible account to use the plugin as far as I know? This is an advantage in my opinion, but generating a shared link via the plugin would require that.

**So +1 for the manual embed approach for now.** :-)

(With regard to my plugin, it is quite incomplete for now: it misses validation for the link, which I think would be necessary not enabling pasting any valid url and embed it.)
